### ISSUE TITLE
Plotyy: PlotyyLabelNum not set

### ISSUE THREAD

**Issue by egeerardyn** [matlab2tikz/matlab2tikz#874]
During the conversion of some of my old FIG files to TikZ, I encountered a problem with one of those.
Now, this might be a very uncommon situation, since the FIG files were generated with an older version of MATLAB (2012a/b, I suppose), yet I call the conversion in R2016a.

As I start from a FIG file, I don't have code to reproduce this and I'm just dumping this here to ensure that we keep track of it.

To reproduce the issue:
 - FIG file is available at https://dl.dropboxusercontent.com/u/452005/m2t/issue874/pdfs.fig

Code used during conversion:

```matlab
files = dir('*.fig');

asTikz = @(file) strrep(file.name, '.fig','.tikz');

for file = files(:)'
    fprintf(' %s \t -> \t %s \n', file.name, asTikz(file));
    
    hFig = open(file.name);
    finallyCloseFigure = onCleanup(@() close(hFig));
    try
        matlab2tikz(asTikz(file), 'figurehandle', hFig, ...
                    'width', '\figurewidth', ...
                    'height', '\figureheight', ...
                    'standalone', false, ...
                    'externalData', true,...
                    'showInfo', false);
    catch ME
        fprintf(2,'Error "%s" during conversion of %s\n%s\n\n', ...
                ME.identifier, file.name, getReport(ME));
    end
end
```

The figure should look something like:
<img width="549" alt="screen shot 2016-03-21 at 11 15 05" src="https://cloud.githubusercontent.com/assets/177360/13916052/89437156-ef58-11e5-99a2-2df07d9d4632.png">

I currently **work around** the issue by means of a patch:
```matlab
function m2t = getPlotyyReferences(m2t, axisHandle)
%% ... help text ...

m2t.PlotyyLabelNum = []; %% ADD THIS LINE AT THE BEGINNING

%% ... other code ...
```
which prevents m2t from crashing, but might have unintended side-effects.
Probably the solution is to be safer about the `m2t.PlotyyLabelNum` variable, e.g. initialize it globally.

**Comment by egeerardyn** [comment]
After #883, this workaround is no longer needed. However, the figure doesn't compile perfectly (at the tip of #883 at the time of writing), since both axes are in the wrong order for LaTeX:
 - the first axis defines the legend (i.e. references some styles)
 - the second axis actually defines the styles:

```latex
% This file was created by matlab2tikz.
%
%The latest updates can be retrieved from
%  http://www.mathworks.com/matlabcentral/fileexchange/22022-matlab2tikz-matlab2tikz
%where you can also make suggestions and rate matlab2tikz.
%
\documentclass[tikz]{standalone}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{pgfplots}
\usepackage{grffile}
\pgfplotsset{compat=newest}
\usetikzlibrary{plotmarks}
\usetikzlibrary{arrows.meta}
\usepgfplotslibrary{patchplots}
\usepackage{amsmath}

\begin{document}
\begin{tikzpicture}

\begin{axis}[%
width=5.511in,
height=4.347in,
at={(0.924in,0.587in)},
scale only axis,
xmode=log,
xmin=10,
xmax=2048,
xminorticks=true,
xlabel={Truncation Time (Samples)},
ymin=0,
ymax=2.5,
ytick={  0, 0.5,   1, 1.5,   2, 2.5},
ylabel={RMS Error},
legend style={legend cell align=left,align=left,draw=black}
]
\addlegendimage{/pgfplots/refstyle=plotyyref:leg-1}\addlegendentry{\$F\$-test};
\addlegendimage{/pgfplots/refstyle=plotyyref:leg0}\addlegendentry{Exponential Fit};
\addplot [color=black,line width=1.5pt]
  table[]{test_plotyy-1.tsv};
\addlegendentry{RMS Error};

\addplot [color=black,line width=1.5pt,mark size=1.7pt,only marks,mark=*,mark options={solid}]
  table[]{test_plotyy-2.tsv};
\addlegendentry{RMS Error};

\end{axis}

\begin{axis}[%
width=5.511in,
height=4.347in,
at={(0.924in,0.587in)},
scale only axis,
xmode=log,
xmin=10,
xmax=2048,
xtick={10,100,1000},
xticklabels={\empty},
xminorticks=true,
ymin=0,
ymax=500,
ytick={  0, 100, 200, 300, 400, 500},
ylabel={\# Realizations},
axis x line*=bottom,
axis y line*=right
]
\addplot[ybar,bar width=6,fill=white!40!black,draw=white!40!black,area legend] plot table[] {test_plotyy-3.tsv};
\addplot[forget plot,color=white!15!black,line width=2.0pt] table[] {test_plotyy-4.tsv};
\label{plotyyref:leg1};

\addplot[ybar,bar width=4,fill=white!70!black,draw=white!70!black,area legend] plot table[] {test_plotyy-5.tsv};
\addplot[forget plot,color=white!15!black,line width=2.0pt] table[] {test_plotyy-6.tsv};
\label{plotyyref:leg2};

\end{axis}
\end{tikzpicture}%
\end{document}
```
